### PR TITLE
Allow skipping of update phases when they were already executed

### PR DESCRIPTION
**Pull request by evgeni**
Fixes: 5c59f9cdd2b7c21174a531fffe8bbdd863029928

### CODE REVIEW

File: lib/foreman_maintain/update_runner.rb [hunk 1]
```diff
@@ -90,6 +94,17 @@ def run_phase(phase)
       @ask_to_confirm_update = phase == :pre_update_checks
     end
 
+    def skip_phase(skipped_phase)
+      with_non_empty_scenario(skipped_phase) do |scenario|
+        @reporter.before_scenario_starts(scenario)
+        @reporter.puts <<~MESSAGE
+          Skipping #{skipped_phase} phase as it was already run before.
+          To enforce to run the phase, use `upgrade run --phase #{skipped_phase}`
```
Comment (evgeni) [inline lib/foreman_maintain/update_runner.rb:102]: This line is obviously nonsense.

File: lib/foreman_maintain/update_runner.rb [hunk 2]
```diff
@@ -110,7 +125,7 @@ def rollback_pre_migrations
     end
 
     def with_non_empty_scenario(phase)
-      next_scenario = scenario(phase)
+      next_scenario = find_scenario(phase)
```
Comment (evgeni) [inline lib/foreman_maintain/update_runner.rb:128]: this was unused, so nobody noticed

File: lib/foreman_maintain/update_runner.rb [hunk 3]
```diff
@@ -90,12 +94,23 @@ def run_phase(phase)
...
+      end
+    end
+
     private
 
     def rollback_pre_migrations
       raise "Unexpected phase #{phase}, expecting pre_migrations" unless phase == :pre_migrations
 
-      rollback_needed = scenario(:pre_migrations).steps.any? { |s| s.executed? && s.success? }
+      rollback_needed = find_scenario(:pre_migrations).steps.any? { |s| s.executed? && s.success? }
```
Comment (evgeni) [inline lib/foreman_maintain/update_runner.rb:113]: I am amazed this was not noticed

**Comment by sayan3296** [comment]
I tested with the packit build on el8, which helps solve the issue.
```
rubygem-foreman_maintain-1.7.4-1.20240926093223538747.pr931.2.g93ae9d8.el8.noarch
```

**Comment by evgeni** [comment]
After discussion with @ares, I am going to merge this to get a release out.

@ehelms you can yell at us next week (or not).

**Pull request merged**